src/app/layout/qgslayoutdesignerdialog.cpp

```diff
@@ -739,6 +740,66 @@ QgsAppLayoutDesignerInterface *QgsLayoutDesignerDialog::iface()
   return mInterface;
 }
 
+static bool cmpByText_( QAction *a, QAction *b )
+{
+  return QString::localeAwareCompare( a->text(), b->text() ) < 0;
+}
+
+QMenu *QgsLayoutDesignerDialog::createPopupMenu()
+{
+  QMenu *menu = QMainWindow::createPopupMenu();
+  QList< QAction * > al = menu->actions();
+  QList< QAction * > panels, toolbars;
+
+  if ( !al.isEmpty() )
+  {
+    bool found = false;
+    for ( int i = 0; i < al.size(); ++i )
+    {
+      if ( al[ i ]->isSeparator() )
+      {
+        found = true;
+        continue;
+      }
+
+      if ( !found )
+      {
+        panels.append( al[ i ] );
+      }
+      else
+      {
+        toolbars.append( al[ i ] );
+      }
+    }
+
+    std::sort( panels.begin(), panels.end(), cmpByText_ );
```

Comment:
Could you make cmpByText_ an lamdba here?

Reply:
Yep, fair enough. If it was just in this function I'd rather it was scoped in here (e.g. 

    auto cmpByText_ = [] {... } 

But that's not an option if it's used elsewhere